plugins/fuse/server/server_fuse.py

```diff
@@ -388,6 +389,9 @@ def unmountServerFuse(name):
 
     :param name: a key within the list of known mounts.
     """
+    # Have python do a garbage collection.  This ensures any file handle that
+    # could be closed really is closed, and therefore won't block unmounting.
+    gc.collect()
```

Comment:
This seems pretty dirty, and relying on implementation details of the interpreter.

Where are the file handles leaking from?

Reply:
Ah, I see. I'm going to read a bit more about `gc.collect` before approving.

Reply:
Are you sure about using the context manager form of `open` not closing when the context manager exits? I would think that would make an explicit call to `f.close()` immediately.

Reply:
No, I'm not completely sure, but any code that does `fh = open(<file>)` without an explicit close will have this situation, too.

I think it is better to have the `gc.collect` than to expect that all code with close file handles promptly.  As I am using the fuse plugin with large_image, I see this after the tile source cache is freed (the tile source destructors DO close the file handles, but those destructors are only called when gc'ed).

I do see one test in `testFunctionOpen` where there is a line `fh = op.open(...)` without a companion `op.release`, which is at least one source of the non-garbage collected file references.

Reply:
Well, my concern is just that this call could be masking / working around bad practices in other parts of the code where we aren't tearing things down explicitly when we should be. If all it takes to actually address this problem is to close our file handles, I'd rather do that than rely on the garbage collector. If it's in some deep and unreachable part of a third party library, maybe this is an acceptable workaround.

Reply:
My current concern is that in large_image, we cache tile sources, which could have a number of open file handles.  When the tile source is discarded from cache (either because the cache has been flushed or because it has been evicted), the tile source reference is abandoned by the generic caching mechanism.  When python GCs the tile source object, its destructor explicitly closes the file handle(s).  I suppose we could overload the `__del__` method of the [LRUCache](http://cachetools.readthedocs.io/en/latest/) that we use, but that is a less general solution than the `gc.collect` method (and perhaps just as hacky).

Reply:
If I understand the use cases for this correctly, this is only something that will generally happen when shutting down the server.  In this case, I don't think it's too terrible.  We certainly wouldn't want to run explicit garbage collections frequently.

I will add though that this won't solve all problems with unmounting... A file handle could be stored in a global variable or in an object that can't be collected.  We can consider this a problem of the code using this plugin, but I feel like this issue is representative of a bigger problem with what is being done in the fuse plugin.  Mounting/unmounting is a system-level operation that really shouldn't be handled by a web server.  My fear is that as we move on to more esoteric use cases, these kinds of problems will continue to occur.

I think this PR is low impact enough that it's okay to go in, but I would like consider the idea of transitioning the management of the fuse mounts to the system daemon before we build too many assumptions on how it currently works.